### DISCUSSION TITLE
SQL Select Statement from Table Links

### DISCUSSION THREAD

**Discussion by Myxomopka666** [ariacom/Seal-Report#99] [Q&A]
Greetings,

I`m trying to select some aggregated values from a table that is already added in LINQ. How I can acheave that ?
Here is an SS from the environemnt:
![image](https://github.com/ariacom/Seal-Report/assets/36057695/11595554-a570-491d-8392-419b10869688)
Regards!

**Comment by ariacom** [comment]
Normally, the aggregation is defined in the model when you drag you element in the Data panel.
However instead of using LINQ to perform a SELECT from a table, you may use a SQL DataSource at least for performance reasons.

**Reply by Myxomopka666** [reply]
Ok ... So I have a model (SQL or Repo Model) with complicated logic and I would like to create charts based on the result out of the Model with the complicated query. How to do that ? I don`t want to make the "complicated" select again but to get the "Aggregated" result out of the first model. In other words (simplified):
1. Model_1 - SELECT * FROM TABLE_1 , TABLE_2 with some joins and etc.
2. Model_2 - SELECT Col1, Col2, Col3 From Model_1 whithout the need to reexecute the entire Model_1 query.

**Comment by ariacom** [comment]
why do you need 2 models to do that (performances, ease for edition, something else) ? 
Here are my remarks:
- The same model can be used to display charts and tables in several views
-  If you defined 2 models generating the same SQL, it will be executed only once by the engine and the ResultTable is shared
- For edition: You can use Reference model to share the model definitions (for their elements or restrictions)
-   You may define Sequence in models, and then, using scripts, you can build another model result based on the first model executed.

**Reply by Myxomopka666** [reply]
Greetings,
Thanks for the relpy ariacom.
I would like to have couple of charts based on the "main"query but since the query is complicated and takes good amount of time I was wondering if it is possible to use same model result in different models.
Let me give you an exaple (again very simplified):
1. Main model : Select 20 columns from T_1, T_2, T_2 with some filters.
2. 3 different charts based on the result from Main model.
- Chart 1- Axis is defined on column_1 and Data elements are 5 different columns
- Chart 2 - Axis is defined on data columns and data elements are 2 columns
- Chart 3 - pie chart of grouping some TOTALS 
In addition it will be much easier for me to use SQL model insted of repo model because of the number of tables/joins/filters/sub selects/transformations and etc. inside the "main" query.

**Comment by ariacom** [comment]
Ok, I understand better your need,
I made a small report small with 1 main model and 2 models with different charts.
I used the smart copy feature to copy elements and restrictions from the main model, so only 1 model is executed.

Check the report: [1model several charts.zip](https://github.com/ariacom/Seal-Report/files/15008146/1model.several.charts.zip)
